### PR TITLE
add tee_technology flag and test for tee_technology flag

### PR DESCRIPTION
**Pull request by Pranjali-2501**
Added --tee_technology flag in attest CLI that specifies the type of hardware the user wants to use for attestation.

### CODE REVIEW

File: cmd/attest.go [hunk 1]
```diff
@@ -69,11 +70,24 @@ hardware and guarantees a fresh quote.
...
+			// Add logic to open other hardware devices when required.
+			switch teeTechnology {
+			case "sev-snp":
+				attestOpts.TEEDevice, err = client.CreateSevSnpDevice()
+				if err != nil {
+					return fmt.Errorf("failed to open sev-snp device: %v", err)
+				}
+			default:
+				// Change the return statement when more devices are added
+				return fmt.Errorf("tee_technology should be sev-snp")
```
Comment (deeglaze) [inline cmd/attest.go:83]: should be one of ["", "sev-snp"]

File: cmd/attest.go [hunk 2]
```diff
@@ -69,11 +70,24 @@ hardware and guarantees a fresh quote.
...
+			}
+			if len(teeNonce) != 0 {
+				attestOpts.TEENonce = teeNonce
+			}
+		} else {
+			if len(teeNonce) != 0 {
```
Comment (deeglaze) [inline cmd/attest.go:89]: else if to about rightward shift

File: cmd/attest.go [hunk 3]
```diff
@@ -156,6 +170,10 @@ func addKeyFlag(cmd *cobra.Command) {
 	cmd.PersistentFlags().StringVar(&key, "key", "AK", "indicates type of attestation key to use <gceAK|AK>")
 }
 
+func addTeeTechnology(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVar(&teeTechnology, "tee_technology", "", "indicates the type of TEE hardware <sev-snp>")
```
Comment (deeglaze) [inline cmd/attest.go:174]: make <sev-snp> text the same as the above error message about valid values.

File: cmd/attest.go (hunk 1 above)
Comment (Pranjali-2501) [reply cmd/attest.go:83]: I have added a constant for the device name and changed the return statement to "tee_technology should be empty or sev-snp"

File: cmd/attest.go (hunk 2 above)
Comment (Pranjali-2501) [reply cmd/attest.go:89]: Instead of else and a if , I'll change it to `else if`

File: cmd/attest.go (hunk 3 above)
Comment (Pranjali-2501) [reply cmd/attest.go:174]: By default it is set to nil, if used then the possible value for now is `"sev-snp"` only.

Comment (Pranjali-2501) [reply cmd/attest.go:178]: If user left the flag empty then it will say ,
`Error: flag needs an argument: --tee_technology`

Comment (Pranjali-2501) [reply cmd/attest.go:178]: Should I change the description to `"indicates the type of TEE hardware, if set to sev-snp"`

**Review by deeglaze — APPROVED** [review]

File: cmd/attest.go [hunk 4]
```diff
@@ -156,6 +174,10 @@ func addKeyFlag(cmd *cobra.Command) {
 	cmd.PersistentFlags().StringVar(&key, "key", "AK", "indicates type of attestation key to use <gceAK|AK>")
 }
 
+func addTeeTechnology(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVar(&teeTechnology, "tee_technology", "", "indicates the type of TEE hardware. Should be empty or sev-snp")
```
Comment (jkl73) [inline cmd/attest.go:178]: Based on the current naming convention for this CLI, it should be "tee-technology"

File: cmd/attest.go [hunk 5]
```diff
@@ -69,12 +76,23 @@ hardware and guarantees a fresh quote.
...
+				}
+			default:
+				// Change the return statement when more devices are added
+				return fmt.Errorf("tee_technology should be empty or %s", SevSnp)
 			}
+			if len(teeNonce) != 0 {
+				attestOpts.TEENonce = teeNonce
+			}
+		} else if len(teeNonce) != 0 {
+			return fmt.Errorf("use of --teenonce requires specifying TEE hardware type with --tee_technology")
```
Comment (jkl73) [inline cmd/attest.go:95]: Could we change "--teenonce" to be "--tee-nonce" for the naming convention? (I understand this flag is created in this PR)

File: cmd/attest.go [hunk 6]
```diff
@@ -69,12 +76,23 @@ hardware and guarantees a fresh quote.
...
 		attestOpts := client.AttestOpts{}
 		attestOpts.Nonce = nonce
-		if len(teeNonce) != 0 {
-			attestOpts.TEENonce = teeNonce
-			attestOpts.TEEDevice, err = client.CreateSevSnpDevice()
-			if err != nil {
-				return fmt.Errorf("failed to collect TEE attestation report: %v", err)
+		if len(teeTechnology) != 0 {
+			// Add logic to open other hardware devices when required.
+			switch teeTechnology {
```
Comment (jkl73) [inline cmd/attest.go:81]: I feel like it is a bit confusing with a switch block inside an if block.
Can you just use the switch block and add the logic under `else if len(teeNonce) != 0 {` as part of the `case ""`?

File: cmd/attest.go (hunk 4 above)
Comment (Pranjali-2501) [reply cmd/attest.go]: I have changed the flag name to `--tee-technology` to make it consistent throughout the file.

File: cmd/attest.go (hunk 5 above)
Comment (Pranjali-2501) [reply cmd/attest.go]: I have changed it to `--tee-nonce`

File: cmd/attest.go (hunk 6 above)
Comment (Pranjali-2501) [reply cmd/attest.go]: As we discussed , I have used a `switch case` instead of `if else` and it is working fine with all possible cases.

**Review by jkl73 — APPROVED** [review]

**Pull request merged**